packages/story-editor/src/components/footer/carousel/karma/carouselDrawer.karma.js

```diff
@@ -0,0 +1,123 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { Fixture } from '../../../../karma';
+
+describe('Carousel Drawer', () => {
+  let fixture;
+
+  beforeEach(async () => {
+    fixture = new Fixture();
+    await fixture.render();
+
+    // We add some content to the first page to make the thumbnail more interesting
+    await fixture.events.click(fixture.editor.library.media.item(0));
+  });
+
+  afterEach(() => {
+    fixture.restore();
+  });
+
+  describe('with a single-page story', () => {
+    it('should show correctly', async () => {
+      // 1 page
+      expect(fixture.editor.footer.carousel.pages.length).toBe(1);
+
+      // Toggle button exists
+      expect(fixture.editor.footer.carousel.toggle).toBeTruthy();
+
+      // No scroll buttons
+      expect(fixture.editor.footer.carousel.next).toBeFalsy();
+      expect(fixture.editor.footer.carousel.previous).toBeFalsy();
+
+      await fixture.snapshot();
+    });
+
+    it('should collapse and expand', async () => {
+      await fixture.events.click(fixture.editor.footer.carousel.toggle);
+      await fixture.events.sleep(800);
+
+      expect(fixture.editor.footer.carousel.pages.length).toBe(0);
+      await fixture.snapshot('Collapsed');
+
+      await fixture.events.click(fixture.editor.footer.carousel.toggle);
+      await fixture.events.sleep(800);
+
+      expect(fixture.editor.footer.carousel.pages.length).toBe(1);
+      await fixture.snapshot('Re-expanded');
+    });
+  });
+
+  describe('with a multi-page story', () => {
+    const EXTRA_PAGES = 9;
+
+    beforeEach(() => {
+      // Let's add some extra pages
+      let pagesToAdd = EXTRA_PAGES;
+
+      const addPage = async () => {
+        if (pagesToAdd <= 0) {
+          return undefined;
+        }
+
+        // Add a page
+        await fixture.events.click(fixture.editor.canvas.pageActions.addPage);
+
+        // Add some media to it
+        await fixture.events.click(
+          fixture.editor.library.media.item(pagesToAdd % 8)
+        );
+
+        // Keep going
+        pagesToAdd -= 1;
+        return addPage();
+      };
+
+      return addPage();
+    });
+
+    it('should show correctly', async () => {
+      // 20 pages
+      expect(fixture.editor.footer.carousel.pages.length).toBe(1 + EXTRA_PAGES);
+
+      // Toggle button exists
+      expect(fixture.editor.footer.carousel.toggle).toBeTruthy();
+
+      // Both scroll buttons exist
+      expect(fixture.editor.footer.carousel.next).toBeTruthy();
+      expect(fixture.editor.footer.carousel.previous).toBeTruthy();
+
+      await fixture.snapshot();
+    });
+
+    it('should collapse and expand', async () => {
+      await fixture.events.click(fixture.editor.footer.carousel.toggle);
+      await fixture.events.sleep(800);
```

Comment:
Nit: Was this based on the animation time? If yes, probably could be reduced a bit now?

Reply:
Fixed in acf6fc10be.